tools/dgeni/src/processors/addLinkTagToDaffodilReferences.ts

```diff
@@ -0,0 +1,23 @@
+import { Processor, Document } from 'dgeni';
+
+/**
+ * Adds a link tag ({@link }) around daffodil models, classes, etc.
+ */
+export class AddLinkTagToDaffodilReferencesProcessor implements Processor {
+	name = 'addLinkTagToDaffodilReferences';
+	$runAfter = ['docs-processed'];
+	$runBefore = ['rendering-docs'];
+
+	$process(docs: Document[]): Document[] {
+		return docs.map(doc => {
+			doc.typeParams = doc.typeParams?.slice(1, doc.typeParams.length-1).replace(/(Daff([A-Z]|[a-z])*)/g, '{@link $1}');
```

Comment:
Are all the interfaces necessarily going to begin with `Daff`? I think no, for example `T extends MagentoProduct`. I guess the others will be added in a different processor later?

Reply:
I think that everything that will be part of daffodil's official public api will start with `Daff`. We had a conversation a few weeks ago about possibly removing the `MagentoProduct` etc. from the api, but you pointed out that the Magento factories we made wouldn't work. But I still think these platform specific models shouldn't be included in our docs, so instead of adding links to platform specific models later, I plan on excluding them from the docs entirely.

Reply:
there are still a few edge cases like `Dict`. Something to keep in mind but it shouldn't hold up this PR

Reply:
That's a good point. I wonder if we should just have a specific tag for things like `Dict`, since I don't think there should be many non-Daff-prefixed models in the future.